test/manipulation.test.js

```diff
@@ -1118,6 +1118,14 @@ describe('manipulation', function() {
       });
     });
 
+    it('fails when id does not exist in db using replaceById', function(done) {
```

Comment:
I am afraid the test for `replaceById` does not belong to `replaceOrCreate` describe block.

It looks like we don't have any explicit tests for `replaceById` yet and test `replaceAttributes` only ([source](https://github.com/strongloop/loopback-datasource-juggler/blob/c145b9359bae974ab2a18e641050109a292abec2/test/manipulation.test.js#L1158)). It's time to start a new `describe` block then! You can add it e.g. after `describe('updateAttributes')` [here](https://github.com/strongloop/loopback-datasource-juggler/blob/c145b9359bae974ab2a18e641050109a292abec2/test/manipulation.test.js#L1348-L1349).